python/pyspark/sql/types.py

```diff
@@ -1599,6 +1599,18 @@ def _has_nulltype(dt: DataType) -> bool:
         return isinstance(dt, NullType)
 
 
+def _has_non_nullable(dt: Optional[Union[AtomicType, StructType]]) -> bool:
```

Comment:
How about just simply:

```python
def _has_non_nullable(dt: DataType) -> bool:
```

I believe it would fix the linter

Reply:
OK, let me fix it.